paddle/lib/sgd_optimizer.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __PADDLE_SGD_OPTIMIZER_H__
+#define __PADDLE_SGD_OPTIMIZER_H__
+
+#include "parameter_optimizer.h"
+#include "training_ops.h"
+
+class SGDOptimizer : public ParameterOptimizer {
+  public:
+  /*! \brief call the applySGD for example  */
+  void update(Tensor<T> &parameter,
+              const Tensor<T> &gradient,
+              double learning_rate) {
+    applyGradientDescent(parameter, gradient, learning_rate);
+  }
+};
+
+class MomentumOptimizer : public ParameterOptimizer {
+
+public:
+  /*! \brief call the applyXX for example  */
+  void update(Tensor<T> &parameter,
```

Comment:
I don't think we should change the update function signature, just use it like this 

fill the `momentum vector`  with `optimizerConfig` when create_optimizer.
just use `update(parameter, gradient, learning_rate) `to call its real update function.

We cannot change the signature because they really have different arguments. 
for example, `RMSProp` algorithm which takes more arguments, it will make the signature more confuse. 
RMSProp see here : http://caffe.berkeleyvision.org/tutorial/solver.html